zebrad/src/components/inbound.rs

```diff
@@ -228,11 +228,20 @@ impl Service<zn::Request> for Inbound {
         match req {
             zn::Request::Peers => {
                 if let Setup::Initialized { address_book, .. } = &self.network_setup {
+                    // # Security
+                    //
                     // We could truncate the list to try to not reveal our entire
                     // peer set. But because we don't monitor repeated requests,
                     // this wouldn't actually achieve anything, because a crawler
                     // could just repeatedly query it.
-                    let mut peers = address_book.lock().unwrap().sanitized();
+                    //
+                    // # Correctness
+                    //
+                    // Briefly hold the address book threaded mutex while
+                    // cloning the address book. Then sanitize after releasing
+                    // the lock.
+                    let peers = address_book.lock().unwrap().clone();
+                    let mut peers = peers.sanitized();
```

Comment:
@fanatid our external address should be added to the cloned peers before they are `sanitized()`.